servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancer.java

```diff
@@ -417,14 +454,16 @@ public void onComplete() {
                         // Failure in selection could be temporary, hence add it to the queue and be consistent
                         // with the fact that select failure does not close a connection.
                         return newCnx.closeAsync().concat(failed(new ConnectionRejectedException(
-                                "Newly created connection " + newCnx + " rejected by the selection filter.")));
+                                "Newly created connection " + newCnx + " for " + targetResource
+                                        + " was rejected by the selection filter.")));
                     }
                     if (host.addConnection(newCnx)) {
                         return succeeded(newCnx);
                     }
-                    return newCnx.closeAsync().concat(this.usedHosts == CLOSED_LIST ? failedLBClosed() :
+                    return newCnx.closeAsync().concat(this.usedHosts == CLOSED_LIST ? failedLBClosed(targetResource) :
                             failed(new ConnectionRejectedException(
-                                    "Failed to add newly created connection for host: " + host)));
+                                    "Failed to add newly created connection for " + targetResource
```

Comment:
I'm not sure which one to choose: `newCnx` or `host`? Or both?
In fairness, the fact that the actual `newCnx` would include the resolved address is an implementation detail of a particular subclass. On `host` we can rely, as it's a private static class here.
Having those stated, it seems to me we should include both the `host` and the `newCnx` in the log. WDYT?

Reply:
Both will be too much because they duplicate the address, and `Host#toString()` contains too much info that is not required for an exception message. Let's pick `newCnx`. We have a test that makes sure its `toString()` is consistent: #1746.

Reply:
After reading your comment in the next discussion and thinking about it one more time, doesn't look like we need to make these two exception messages consistent. The "Newly created connection rejected" is rejected by the selector and doesn't need to tell anything about the host. "Failed to add newly created connection for host" is rejected by the host, it's better to include both (`newCtx` and `host`) in this case to let users know the connection id and the `host` state.